js/src/name-pubsub/cancel.js

```diff
@@ -63,24 +52,30 @@ module.exports = (createCommon, options) => {
 
     it('should cancel a subscription correctly returning true', function (done) {
       this.timeout(300 * 1000)
-      const ipnsPath = `/ipns/${nodeId}`
-
-      series([
-        (cb) => ipfs.name.pubsub.subs(cb),
-        (cb) => ipfs.name.publish(value, { resolve: false }, cb),
-        (cb) => ipfs.name.resolve(nodeId, cb),
-        (cb) => ipfs.name.pubsub.subs(cb),
-        (cb) => ipfs.name.pubsub.cancel(ipnsPath, cb),
-        (cb) => ipfs.name.pubsub.subs(cb)
-      ], (err, res) => {
+      const id = 'QmNP1ASen5ZREtiJTtVD3jhMKhoPb1zppET1tgpjHx2NGA'
+      const ipnsPath = `/ipns/${id}`
+
+      ipfs.name.pubsub.subs((err, res) => {
         expect(err).to.not.exist()
-        expect(res).to.exist()
-        expect(res[0]).to.eql([]) // initally empty
-        expect(res[4]).to.have.property('canceled')
-        expect(res[4].canceled).to.eql(true)
-        expect(res[5]).to.be.an('array').that.does.not.include(ipnsPath)
+        expect(res).to.eql([]) // initally empty
+
+        ipfs.name.resolve(id, (err) => {
+          expect(err).to.exist()
+          series([
+            (cb) => ipfs.name.pubsub.subs(cb),
+            (cb) => ipfs.name.pubsub.cancel(ipnsPath, cb),
+            (cb) => ipfs.name.pubsub.subs(cb)
+          ], (err, res) => {
+            expect(err).to.not.exist()
+            expect(res).to.exist()
+            expect(res[0]).to.be.an('array').that.does.include(ipnsPath)
+            expect(res[1]).to.have.property('canceled')
+            expect(res[1].canceled).to.eql(true)
+            expect(res[2]).to.be.an('array').that.does.not.include(ipnsPath)
```

Comment:
A note for the future, if you use `async/auto` instead you can give the results names. That way if we add another thing to the pipeline for this we don't have to update all our array indexes.

Reply:
Thanks! Good tip, I changed it